stdlib/experimental/geo/geo.flux

```diff
@@ -138,14 +658,125 @@ gridFilter = (
+//   
+//   Specify record properties for the shape. 
+// 
+// - tables: Input data. Default is piped-forward data (`<-`).
+// 
+// ## Examples
+// ### Strictly filter data to a specified region
+// ```
+// # import "array"
+// import "experimental/geo"
+// # 
+// # data = array.from(
+// #     rows: [
+// #         {_time: 2021-01-01T00:00:00Z, id: "a213b", _field: "latitude", _value: 41.01433},
+// #         {_time: 2021-01-02T01:00:00Z, id: "a213b", _field: "latitude", _value: 40.9228},
+// #         {_time: 2021-01-03T02:00:00Z, id: "a213b", _field: "latitude", _value: 39.08433},
+// #         {_time: 2021-01-01T00:00:00Z, id: "a213b", _field: "longitude", _value: -70.7515},
+// #         {_time: 2021-01-02T01:00:00Z, id: "a213b", _field: "longitude", _value: -73.3527},
+// #         {_time: 2021-01-03T02:00:00Z, id: "a213b", _field: "longitude", _value: -75.9978},
+// #     ],
+// # )
+// #     |> group(columns: ["id", "_field"])
+// #     |> geo.shapeData(latField: "latitude", lonField: "longitude", level: 5)
+// 
+// < data
+// >     |> geo.strictFilter(region: {lat: 40.69335938, lon: -73.30078125, radius: 50.0})
+// ```
+// 
+// tags: transformations,filters,geotemporal
+// 
 strictFilter = (tables=<-, region) => tables
     |> filter(fn: (r) => ST_Contains(region: region, geometry: {lat: r.lat, lon: r.lon}))
 
-// Two-phase filtering by specified region.
-// Checks to see if data is already pivoted and contains a lat column.
-// Returns pivoted data.
+// filterRows filters data by a specified geographic region with the option of strict filtering.
+// 
+// This function is a combination of `geo.gridFilter()` and `geo.strictFilter()`.
+// Input data must include an `s2_cell_id` column that is **part of the group key**.
+// 
+// ## Parameters
+// - region: Region containing the desired data points.
+//   
+//   Specify record properties for the shape. 
+// 
+// - minSize: Minimum number of cells that cover the specified region.
+//   Default is `24`.
+// - maxSize: Maximum number of cells that cover the specified region.
+//   Default is `-1` (unlimited).
+// - level: [S2 cell level](https://s2geometry.io/resources/s2cell_statistics.html)
+//   of grid cells. Default is `-1`.
+// 
+//   **Note:** `level` is mutually exclusive with `minSize` and `maxSize` and
+//   must be less than or equal to `s2cellIDLevel`.
+// 
+// - s2cellIDLevel: [S2 cell level](https://s2geometry.io/resources/s2cell_statistics.html)
+//   used in the `s2_cell_id` tag. Default is `-1` (detects S2 cell level from the `s2_cell_id` tag).
+// - strict: Enable strict geographic data filtering. Default is `true`.
+// 
+//   Strict filtering returns only points with coordinates in the defined region.
+//   Non-strict filtering returns all points from S2 grid cells that are partially
+//   covered by the defined region.
+// 
+// - tables: Input data. Default is piped-forward data (`<-`).
+// 
+// ## Examples
+// 
+// ### Strictly filter geotemporal data by region
```

Comment:
This test seems to break on master

```
 error calling function "filterRows" @18:8-18:84: error calling function "getColumn" @geo.flux|161:12-161:39: cannot find column _value
```

Reply:
@Marwes it's a flaky test. If rerun, it may pass. I don't know the exact cause of the flakiness, but I've made a change to the function in https://github.com/influxdata/flux/pull/4340 that may fix it 🤞🏻. I can't reproduce the error locally, but I have seen this error in CI.

Reply:
Strange, I see it consistently locally.

Reply:
Interesting... 🤔